scripts/Muon/GUI/Common/contexts/muon_group_pair_context.py

```diff
@@ -213,7 +213,7 @@ def add_pair(self, pair):
             raise ValueError('Groups and pairs must have unique names')
 
     def show(self, name, run):
-        self[name].show(str(run))
+        self[name].show(tuple(run))
```

Comment:
Could you help me understand the significance of the tuple? As far as I can see it only ever has a single element in it?

The `tuple(run)` transformation seems to be repeated a fair bit? Could we be missing an opportunity to promo `run` to a type?

Reply:
There is a rarely used option to co-add a series of runs upon loading. This has added a complication to the rest of the code whereby when keeping track of which runs a workspace corresponds to it can be one to many. This is currently done by passing around a list of workspaces which is tupled here to produce a hashable object to act as a dict key.

We could promote run into a type in it's own right which would resolve the hashing problem and allow us to encode how to transform it into a string.

Reply:
I've added a new class (see commit message for details), let me know if this wasn't what you meant.